packages/api/src/module/write/email-confirmation/domain/requestEmailConfirmation.spec.ts

```diff
@@ -0,0 +1,22 @@
+import { requestEmailConfirmationCommand } from '@/module/commands/request-email-conformation';
+
+import { requestEmailConfirmation } from './requestEmailConfirmation';
+
+describe('Request email confirmation', () => {
+  it('create event EmailConfirmationWasRequested when requestEmailConfirmation is invoked', () => {
+    const exampleUserId = 'ca63d023-4cbd-40ca-9f53-f19dbb19b0ab';
+    const exampleConfirmationToken = 'sbAPITNMsl2wW6j2cg1H2A';
+    const exampleConfirmationFor = 'user-registration';
+
+    const command = requestEmailConfirmationCommand({
+      userId: exampleUserId,
+      confirmationToken: exampleConfirmationToken,
+      confirmationFor: exampleConfirmationFor,
+    });
+    const result = requestEmailConfirmation(command)();
+
+    expect(result).toHaveLength(1);
```

Comment:
It'd be better to check the whole result. Let's check out `given not generated before, then should be generated` test. 
Because now you haven't checked for example if there is no more event, or if event had properties that should not have (is TypeScript, so even if type doesn't allow that someone may be passed something more).

Reply:
AH, ok - you have `toHaveLength`, but I'm still in favor of checking equality. You can, because it's pure function (no side effects, output based only on input), so it's very easy to just check if result is as expected. It's more readable than splitting checks for different properties.

Reply:
Yes, and my question is why don't just put in with one expect with whole expected output?

Reply:
I don't know 🤷‍♀️ 
Now it is in one expect :)

Reply:
For me it's easier to see whole object if there is test fail :) You have better context to understand problem.